src/data/store/Layers.js

```diff
@@ -156,6 +156,47 @@ Ext.define('GeoExt.data.store.Layers', {
         }
     },
 
+    /**
+     * Bind the layer to the record and initialize synchronized values.
+     *
+     * @param {ol.layer.Base} layer The layer.
+     * @param {Ext.data.Model} [record] The record, if not set it will be
+     *      searched for.
+     */
+    bindLayer: function(layer, record) {
+        var me = this;
+        layer.on('propertychange', me.onChangeLayer, me);
+        if (record === undefined) {
+            record = this.getRecordForLayer(layer);
+        }
+        Ext.Array.forEach(record.synchronizedProperties,
```

Comment:
`record` could be `undefined` here and therefore raise an exception, right?

Reply:
No this cannot really happen as the function is always called in a context where a record exists for the layer. But this could be changed to make it more clear.

Reply:
You are right actually, this was because I used the `changeLayerFilterFn` at times where it should not get used.

I created this follow up PR where I fixed this: https://github.com/geoext/geoext/pull/688

Reply:
Great, thanks